05-html-css/css-grid-intro.md

```diff
@@ -0,0 +1,334 @@
+- Are the rows are in the layout total? How many are defined on the _grid container_?
+  - Are all rows equally sized? In CSS Grid, not every row needs to be the same height
+- For three different grid items, answer the following:
+  - What column line # does this grid item start at?
+  - What column line # does this grid item end at?
+  - How many columns does this grid item span?
+  - What row line # does this grid item start at?
+  - What row line # does this grid item end at?
+  - How many rows does this grid item span?
+
+### Observe
+
+Let's look at a more realistic example. Let's say we've received this wireframe for a website to create. What is the grid container? What are the rows? What are the columns? Are all columns equally sized? Are all rows equally sized? What is each grid item, and how many rows and columns do each one span?
+
+![Realistic CSS Grid Layout](./imgs/css-grid-realistic-example.png)
+
+Let's take a look at one way we'd likely make write this HTML, and its accompanying CSS to set it up to use CSS Grid.
+
+```html
+<!-- index.html -->
+<!-- stuff like the opening html tag, head tag, link to stylesheet, opening body tag -->
+
+<div class="container">
+  <header class="header">
+
+  </header>
+  <nav class="nav">
+
+  </nav>
+  <div class="content">
+
+  </div>
+  <footer class="footer">
+
+  </footer>
+</div>
+
+<!-- close body tag and html tag -->
+```
+
+```css
+/* style.css */
+
+.container {
+  /* these are just an arbitrary widths and heights to look good on our browser */
+  width: 30vw;
+  height: 60vh;
+
+  display: grid;
+}
+
+.header {
+  background-color: #d55e00;
+}
+
+.nav {
+  background-color: #f0e442;
+}
+
+.content {
+  background-color: #009e73;
+}
+
+.footer {
+  background-color: #0072b2;
+}
+```
+
+
+We've set up CSS Grid using `display: grid;` on our grid container, but we don't see it working yet.
+
+## Grid Template
+
+We need to start with broad strokes. In above examples, we identified how many columns and rows each grid item spanned, where they started and where they ended. But we need to first define the following in the context of the entire grid container:
+
+- How many columns are there?
+- What size is each column?
+- How many rows are there?
+- What size is each row?
+
+We answer all of those questions with any property beginning with the phrase "grid-template": `grid-template`, `grid-template-columns`, `grid-template-rows`, and/or `grid-template-areas`.
+
+### Defining Grid Template
+
+For the following section we will be using [this example on Codepen.](https://codepen.io/adadev/pen/bvbKQX?editors=1100) Follow along, and comment and uncomment sections as we go through.
+
+#### Grid Template Columns, Grid Template Rows
+
+To define how many columns there are and what size each column is, you use the CSS attribute `grid-template-columns` so that _it applies to your grid container_. For every column, you provide a value for width, separated by **spaces, with no punctuation**.
```

Comment:
"_it applies to your grid container_" is in italics here, but I'm not sure why. What else would it be, what pitfall are we avoiding?

Reply:
to apply to the entire grid container (as a template), rather than how many columns a specific grid item takes up